script.keymap/resources/lib/editor.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014 Thomas Amland
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from kodi_six import xbmcaddon
+from threading import Timer
+from collections import OrderedDict
+from kodi_six.xbmcgui import Dialog, WindowXMLDialog
+from resources.lib.actions import ACTIONS, WINDOWS
+from resources.lib.utils import tr
+
+
+class Editor(object):
+    def __init__(self, defaultkeymap, userkeymap):
+        """Create the editor object."""
+        self.defaultkeymap = defaultkeymap
+        self.userkeymap = userkeymap
+        self.dirty = False
+
+    def start(self):
+        while True:
```

Comment:
Forever running loops should check for abortion. I'd use xbmc.abortRequested flag for Leia and monitor.abortRequested() for matrix

Reply:
fixed in 1a6d491